### DISCUSSION TITLE
Adding a new language Kazakh (Kazakhstan)

### DISCUSSION THREAD

**Discussion by kovalevcon** [nvaccess/nvda#16139] [Ideas]
Greetings!
I am in Kazakhstan, our native language is [Kazakh](https://ru.wikipedia.org/wiki/%D0%9A%D0%B0%D0%B7%D0%B0%D1%85%D1%81%D0%BA%D0%B8%D0%B9_%D1%8F%D0%B7%D1%8B%D0%BA)
I would like to translate the NVDA program into Kazakh.
Can you tell me what needs to be done for this, preferably step by step?
I am a PHP developer myself, I understand the general syntax of Python, if you tell me how to add a new language, then I will be able to translate the text and make my suggestions.
As I understand it, you need to create a language by analogy like [here](https://github.com/nvaccess/nvda/tree/master/source/locale)
Correct me, tell me what needs to be done.

**Comment by hwf1324** [comment]
Hi, this is just a quick unofficial reply, NV Access is in the process of migrating NVDA translations to Crowdin, new languages may not be added until the migration is complete, but the migration is definitely done after the 2024.1 release. At that time, if you are interested in doing translations, I think everyone would be welcome.

**Reply by hwf1324** [reply]
Related discussion: https://github.com/nvaccess/nvda/discussions/15014

**Comment by zstanecic** [comment]
Ple ase be patient after the version 2.24.1 version. The NVDA is going to be translated using Crowdin. 

When all proces stabilizes and streamlines, it will be open for new languages.

Thanks!

 

 

From: Constantine ***@***.***> 
Sent: Wednesday, February 7, 2024 9:43 AM
To: nvaccess/nvda ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [nvaccess/nvda] Adding a new language Kazakh (Kazakhstan) (Discussion #16139)

 

Greetings!
I am in Kazakhstan, our native language is Kazakh <https://ru.wikipedia.org/wiki/%D0%9A%D0%B0%D0%B7%D0%B0%D1%85%D1%81%D0%BA%D0%B8%D0%B9_%D1%8F%D0%B7%D1%8B%D0%BA> 
I would like to translate the NVDA program into Kazakh.
Can you tell me what needs to be done for this, preferably step by step?
I am a PHP developer myself, I understand the general syntax of Python, if you tell me how to add a new language, then I will be able to translate the text and make my suggestions.
As I understand it, you need to create a language by analogy like here <https://github.com/nvaccess/nvda/tree/master/source/locale> 
Correct me, tell me what needs to be done.

**Reply by kovalevcon** [reply]
It turns out that after the introduction of Crowdin, the Kazakh language will appear for selection in the NVDA application?
Or will it still be necessary to add new localizations as they were before?

**Comment by kovalevcon** [comment]
It turns out that after the introduction of Crowdin, the Kazakh language will appear for selection in the NVDA application?
Or will it still be necessary to add new localizations as they were before?

**Comment by CyrilleB79** [comment]
Hi @kovalevcon

The process is not yet well established. There have been other requests to add a new language that are pending due to the translation framework migration (Khmer / Cambodian, Estonian, Azerbaijani).

I'd recommend, if not already done, that you already subscribe to the [translators mailing list](https://groups.io/g/nvda-translations/topics).
There, you will be able to ask for help about the translation process when it's time.

Cheers,

Cyrille



After NVDA 2024.1 is released, you and other translators of new languages may ask there how to integrate a new language in NVDA. You will probably only be able to translate the interface which is the most important file anyway. Indeed, the migration of the translation of other files (User Guide, Change Log) has not been done for 2024.1 and will probably re-targeted for 2024.2.

**Reply by kovalevcon** [reply]
Hi @CyrilleB79
I see the release of version 2024.1 has taken place, now can I add a new language Kazakh?
I need to get instructions or a recommendation from you on how to do this.

**Comment by CyrilleB79** [comment]
Hi

As I have told you, the documentation translation migration is still on-going since it has been retargeted for 2024.2.

There has not been any news from NV Access regarding the introduction of new languages. I do not know if they plan to include them or not for 2024.2.
@seanbudd, @gerald-hartig or @michaelDCurran could you provide an update here as well as on the translators mailing list?
As of today, there has been request for the following new languages: Khmer / Cambodian, Estonian, Azerbaijani and Kazakh.
Thanks.

@kovalevcon, in the meantime, have you subscribed to the translators mailing list, so that you get the updated information and the instructions when possible?

**Comment by gerald-hartig** [comment]
If those languages are supported we will enable them in Crowdin today. This means that the translators would be able to start work this week. Work would start on the user guide & documentation after 2024.2 is released in June.

**Comment by CyrilleB79** [comment]
@gerald-hartig please send a message on the translators mailing list when translators of these new languages can begin working on it.

Also FYI, the possibility to have a new Serbian (Cyrillic) translation is being discussed [here](https://github.com/nvaccess/nvda/discussions/16465).

**Comment by gerald-hartig** [comment]
This is done https://groups.io/g/nvda-translations/message/4038
